### PR TITLE
Use time instead of duration for timedwait

### PR DESCRIPTION
**Pull request by julianoes**
This changes the interface of `waitOnSignal` from a time duration for
the timeout to an absolute time (timespec).

This makes more sense because that's how the underlying
`pthread_cond_timedwait` is implemented, and this also gets rid of a
race condition which can occur if the time is updated right after the
time is read using `offsetTime`. This made no difference until now but
with the lockstep SITL implementation this can mean that a new time is
set just before we wait for the timeout. In this case, the timeout will
never trigger because SITL waits for an actuator control message (the
next "lockstep") in order to continue and update the time again.

By making this an absolute time, we don't have this race and would just
return the `pthread_cond_timedwait` straightaway if the time has been
updated in the meantime.

### CODE REVIEW

**Comment by julianoes** [comment]
@bkueng I think this is ready for a review, thanks!

File: framework/src/SyncObj.cpp [hunk 1]
```diff
@@ -38,7 +38,7 @@
 #include "DriverFramework.hpp"
 #include "SyncObj.hpp"
 
-#if defined(__PX4_POSIX) || defined(__PX4_QURT)
+#if defined(__PX4_POSIX_SITL)
```
Comment (bkueng) [inline framework/src/SyncObj.cpp]: Is this a new define?

File: framework/src/Time.cpp [hunk 2]
```diff
@@ -69,15 +73,20 @@ using namespace DriverFramework;
...
 #else
-#if defined(__PX4_POSIX) || defined(__PX4__QURT)
-	return px4_clock_gettime(CLOCK_MONOTONIC, &ts);
-#else
-	return clock_gettime(CLOCK_MONOTONIC, &ts);
+
+#ifndef CLOCK_MONOTONIC
+#define CLOCK_MONOTONIC 1
 #endif
+	// For anything PX4, use monotonic and call use PX4 define.
```
Comment (bkueng) [inline framework/src/Time.cpp]: `call use PX4 define` is not very clear

File: framework/src/SyncObj.cpp (hunk 1 above)
Comment (julianoes) [reply framework/src/SyncObj.cpp]: Yes.
https://github.com/PX4/Firmware/pull/10648/commits/ad81253054a81d6c09ac3b38d95479b91fcd806a

File: framework/src/Time.cpp (hunk 2 above)
Comment (julianoes) [reply framework/src/Time.cpp]: Thx, I'll fix it.

File: framework/src/SyncObj.cpp (hunk 1 above)
Comment (bkueng) [reply framework/src/SyncObj.cpp]: Ok. Can't we always define it for `__PX4_POSIX` as `px4_pthread_cond_timedwait` and the Firmware will then map it to the right thing?

**Pull request merged**